### PR TITLE
[android-auto] Remove android:foregroundServiceType="location" from CarAppService

### PR DESCRIPTION
**Pull request by AndrewShkrob**
Closes: https://github.com/organicmaps/organicmaps/issues/7263

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -436,7 +436,6 @@
         android:name="app.organicmaps.MapPlaceholderActivity"/>
     <service
         android:name="app.organicmaps.car.CarAppService"
-        android:foregroundServiceType="location"
```
Comment (biodranik) [inline android/app/src/main/AndroidManifest.xml:439]: Is it a correct fix? Will AA work properly when the app is in the background on the device (or the device is locked)? Someone from Google describes a different fix here: https://issuetracker.google.com/issues/322622698#comment29

Comment (AndrewShkrob) [reply android/app/src/main/AndroidManifest.xml:439]: This is exactly what I've done. The rest part is already implemented on our side.

**Comment by biodranik** [comment]
@rtsisyk was it about normal (app) service or AA service (this change)?

**Comment by AndrewShkrob** [comment]
> As far as I remember, the foreground service didn't start on Android 14 without this line. Has anything changed?

I've rechecked the docs, and there is no information about it. It was probably a misunderstanding. However, we need to test it to be 100% sure.

**Comment by Jean-BaptisteC** [comment]
I can test on Android 14, leave me more time

**Comment by AndrewShkrob** [comment]
> Did you have a chance to test it on Android 14? The only device I have with Android Auto is on Android 12.

No, I don't have Android 14.

**Comment by Jean-BaptisteC** [comment]
I continue to have crash with latest version of AA :(
```
Process: app.organicmaps.web.debug, PID: 12092
java.lang.RuntimeException: android.app.MissingForegroundServiceTypeException: Starting FGS without a type  callerApp=ProcessRecord{2defb4c 12092:app.organicmaps.web.debug/u0a318} targetSDK=34
at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$0(RemoteUtils.java:156)
at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda3.run(D8$$SyntheticClass:0)
at android.os.Handler.handleCallback(Handler.java:959)
at android.os.Handler.dispatchMessage(Handler.java:100)
at android.os.Looper.loopOnce(Looper.java:232)
at android.os.Looper.loop(Looper.java:317)
at android.app.ActivityThread.main(ActivityThread.java:8592)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: android.app.MissingForegroundServiceTypeException: Starting FGS without a type  callerApp=ProcessRecord{2defb4c 12092:app.organicmaps.web.debug/u0a318} targetSDK=34
at android.app.MissingForegroundServiceTypeException$1.createFromParcel(MissingForegroundServiceTypeException.java:53)
at android.app.MissingForegroundServiceTypeException$1.createFromParcel(MissingForegroundServiceTypeException.java:49)
at android.os.Parcel.readParcelableInternal(Parcel.java:5016)
at android.os.Parcel.readParcelable(Parcel.java:4998)
at android.os.Parcel.createExceptionOrNull(Parcel.java:3178)
at android.os.Parcel.createException(Parcel.java:3167)
at android.os.Parcel.readException(Parcel.java:3150)
at android.os.Parcel.readException(Parcel.java:3092)
at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6960)
at android.app.Service.startForeground(Service.java:776)
at app.organicmaps.car.CarAppService.onCreateSession(CarAppService.java:56)
at androidx.car.app.CarAppBinder.lambda$onAppCreate$0$androidx-car-app-CarAppBinder(CarAppBinder.java:115)
at androidx.car.app.CarAppBinder$$ExternalSyntheticLambda4.dispatch(D8$$SyntheticClass:0)
at androidx.car.app.utils.RemoteUtils.lambda$dispatchCallFromHost$0(RemoteUtils.java:151)
at androidx.car.app.utils.RemoteUtils$$ExternalSyntheticLambda3.run(D8$$SyntheticClass:0) 
at android.os.Handler.handleCallback(Handler.java:959) 
at android.os.Handler.dispatchMessage(Handler.java:100) 
at android.os.Looper.loopOnce(Looper.java:232) 
at android.os.Looper.loop(Looper.java:317) 
at android.app.ActivityThread.main(ActivityThread.java:8592) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878) 
```

**Comment by biodranik** [comment]
@Jean-BaptisteC does it crash independently of changes in this PR? Or only when you run it with this PR?

Could it be caused by web-debug build? Does this google-play signed apk work? (it will replace the Google Play version) https://cdn.organicmaps.app/apk/OrganicMaps-24081521-Google-Play.apk

**Comment by Jean-BaptisteC** [comment]
Your APK contains always `android:foregroundServiceType="location"` for `CarAppService`
I have tried with beta from 9 august on Firebase tester, I have the same crash

**Comment by AndrewShkrob** [comment]
I'd suggest to use this type of foreground service:
https://developer.android.com/about/versions/14/changes/fgs-types-required#special-use

**Comment by biodranik** [comment]
What kind of explanation should be provided to reviewers?

Why do you think it will work properly?

**Comment by AndrewShkrob** [comment]
> What kind of explanation should be provided to reviewers?

"It's an AA service. Since Android 14 every FGS must have a type. There is no specific service type for AA. We can't use location type as we can't be sure the user has given the required permissions before the service starts."

> Why do you think it will work properly?

We must specify the service type. We will specify it. We shouldn't see a `MissingForegroundServiceTypeException` after that.

**Comment by AndrewShkrob** [comment]
> Are there any downsides of keeping this attribute for now? The code does crash without it.

The app will crash if the user hasn't granted location permissions before launching Android Auto.

**Comment by AndrewShkrob** [comment]
> > > Are there any downsides of keeping this attribute for now? The code does crash without it.
> > 
> > 
> > The app will crash if the user hasn't granted location permissions before launching Android Auto.
> 
> OK. It is probably less harmful than crashing every time like with this patch :-) Let's wait while Google fixes that.

I have updated the patch with proper service type.

**Comment by biodranik** [comment]
@AndrewShkrob what are the exact steps to reproduce the issue? Can it be reproduced/tested locally? Did you test it?

**Comment by AndrewShkrob** [comment]
> @AndrewShkrob what are the exact steps to reproduce the issue? Can it be reproduced/tested locally? Did you test it?

What do you want to reproduce?

* No FGS type specified -> the app will crash
* `location` type for FGS is used and no permissions are granted before starting the service -> the app will crash

With this patch, it will work in both cases

File: android/app/src/main/res/values/donottranslate.xml [hunk 2]
```diff
@@ -63,4 +63,6 @@
   <string name="auto_enum_value" translatable="false">AUTO</string>
   <string name="placepage_behavior" translatable="false">com.google.android.material.bottomsheet.BottomSheetBehavior</string>
   <string name="car_notification_channel_name" translatable="false">CAR_NOTIFICATION_CHANNEL</string>
+
+  <string name="android_auto_fgs_explanation_for_special_use" translatable="false">Since Android 14, every Foreground Service must have a type. There is no specific service type for Android Auto.</string>
```
Comment (biodranik) [inline android/app/src/main/res/values/donottranslate.xml:67]: Let's also add a link to the original issue: https://issuetracker.google.com/issues/322622698

**Review by biodranik — APPROVED** [review]
I'm ok to merge it, just curious if _anyone_ has tested it at least once, and is testing on the Mac locally is enough.

**Review by biodranik — APPROVED** [review]
It will hang forever without anyone's involvement. So I'm merging it into the master to force more active testing.

**Pull request merged**